### PR TITLE
fix v2.1 t2i sampling v-prediction

### PR DESCRIPTION
**Pull request by DongyaoZhu**

### CODE REVIEW

File: examples/stable_diffusion_v2/ldm/models/diffusion/dpm_solver/sampler.py [hunk 1]
```diff
@@ -29,6 +29,7 @@ def __init__(self, model, algorithm_type="dpmsolver", **kwargs):
         self.algorithm_type = algorithm_type
         self.register_buffer("alphas_cumprod", model.alphas_cumprod)
         self.noise_schedule = NoiseScheduleVP("discrete", alphas_cumprod=self.alphas_cumprod)
+        self.model_type = kwargs.get("model_type", "noise")
```
Comment (congw729) [inline examples/stable_diffusion_v2/ldm/models/diffusion/dpm_solver/sampler.py:32]: It seems “noise” is not a key in kwargs ?

File: examples/stable_diffusion_v2/text_to_image.py [hunk 2]
```diff
@@ -156,15 +156,18 @@ def main(args):
         lora_only_ckpt=args.lora_ckpt_path,
     )
 
+    commom_kwargs = {
+        "model_type": "v" if args.version == "2.1" else "noise",
+    }
```
Comment (congw729) [inline examples/stable_diffusion_v2/text_to_image.py:161]: According to this file, “model_type” is the only key in commom_kwargs. BTW, what’s the meaning of commom? Common?

File: examples/stable_diffusion_v2/ldm/models/diffusion/dpm_solver/sampler.py (hunk 1 above)
Comment (DongyaoZhu) [reply examples/stable_diffusion_v2/ldm/models/diffusion/dpm_solver/sampler.py:32]: python syntactic sugar

File: examples/stable_diffusion_v2/text_to_image.py (hunk 2 above)
Comment (DongyaoZhu) [reply examples/stable_diffusion_v2/text_to_image.py:161]: typo fixed; wrapping with kwargs to be extendable

File: examples/stable_diffusion_v2/ldm/models/diffusion/dpm_solver/sampler.py (hunk 1 above)
Comment (SamitHuang) [inline examples/stable_diffusion_v2/ldm/models/diffusion/dpm_solver/sampler.py:32]: Suggest follow the naming in diffuser. `model_type` -> `prediction_type`
ref: https://huggingface.co/docs/diffusers/api/schedulers/ddpm#diffusers.DDPMScheduler.prediction_type

File: examples/stable_diffusion_v2/text_to_image.py [hunk 3]
```diff
@@ -156,15 +156,18 @@ def main(args):
         lora_only_ckpt=args.lora_ckpt_path,
     )
 
+    dpm_common_kwargs = {
+        "model_type": "v" if args.version == "2.1" else "noise",
```
Comment (SamitHuang) [inline examples/stable_diffusion_v2/text_to_image.py:160]: Not all sd 2.1 variants use v-prediciton. 
2.0-v: v_prediction. 
2.1-base: noise
2.1-v: v_prediction

We may follow the naming rule in hf. if it is not base 512 model, we add postfix like _768_v to version number, e.g. 2.0_768_v.  And judge prediction type by ending with "_v"

**Pull request merged**